p2p/transport/quic/transport.go

```diff
@@ -446,13 +478,20 @@ func (t *transport) Proxy() bool {
 
 // Protocols returns the set of protocols handled by this transport.
 func (t *transport) Protocols() []int {
-	return []int{ma.P_QUIC}
+	if t.enableDraft29 {
+		return []int{ma.P_QUIC, ma.P_QUIC_V1}
+	}
+	return []int{ma.P_QUIC_V1}
 }
 
 func (t *transport) String() string {
 	return "QUIC"
 }
 
 func (t *transport) Close() error {
-	return t.connManager.Close()
+	// We may register multiple protocols, but they are all the same transport, so only support getting closed once.
+	t.closeOnce.Do(func() {
```

Comment:
Why is this needed? Can't `connManager.Close` be called multiple times?

Reply:
no it can't be. Should it be?

Reply:
Let's try to dedupe this at the higher level